src/library-authoring/collections/CollectionDetails.tsx

```diff
@@ -0,0 +1,177 @@
+
+interface BlockCountProps {
+  count: number,
+  blockType?: string,
+  label: React.ReactNode,
+  className?: string,
+}
+
+const BlockCount = ({
+  count,
+  blockType,
+  label,
+  className,
+}: BlockCountProps) => {
+  const icon = blockType && getItemIcon(blockType);
+  return (
+    <Stack className={classNames('text-center', className)}>
+      <span className="text-muted">{label}</span>
+      <Stack direction="horizontal" gap={1} className="justify-content-center">
+        {icon && <Icon src={icon} size="lg" />}
+        <span>{count}</span>
+      </Stack>
+    </Stack>
+  );
+};
+
+interface CollectionStatsWidgetProps {
+  libraryId: string,
+  collectionId: string,
+}
+
+const CollectionStatsWidget = ({ libraryId, collectionId }: CollectionStatsWidgetProps) => {
+  const { data: blockTypes } = useGetBlockTypes([
+    `context_key = "${libraryId}"`,
+    `collections.key = "${collectionId}"`,
+  ]);
+
+  if (!blockTypes) {
+    return null;
+  }
+
+  const blockTypesArray = Object.entries(blockTypes)
+    .map(([blockType, count]) => ({ blockType, count }))
+    .sort((a, b) => b.count - a.count);
+
+  const totalBlocksCount = blockTypesArray.reduce((acc, { count }) => acc + count, 0);
+  // Show the top 3 block type counts individually, and splice the remaining block types together under "Other".
+  const numBlockTypesShown = 3;
+  const otherBlocks = blockTypesArray.splice(numBlockTypesShown);
+  const otherBlocksCount = otherBlocks.reduce((acc, { count }) => acc + count, 0);
+
+  if (totalBlocksCount === 0) {
+    return (
+      <div
+        className="text-center text-muted align-content-center"
+        style={{
+          height: '72px', // same height as the BlockCount component
+        }}
+      >
+        <FormattedMessage {...messages.detailsTabStatsNoComponents} />
+      </div>
+    );
+  }
+
+  return (
+    <Stack direction="horizontal" className="p-2 justify-content-between" gap={2}>
+      <BlockCount
+        label={<FormattedMessage {...messages.detailsTabStatsTotalComponents} />}
+        count={totalBlocksCount}
+        className="border-right"
+      />
+      {blockTypesArray.map(({ blockType, count }) => (
+        <BlockCount
+          key={blockType}
+          label={<BlockTypeLabel type={blockType} />}
+          blockType={blockType}
+          count={count}
+        />
+      ))}
+      {otherBlocks.length > 0 && (
+        <BlockCount
+          label={<FormattedMessage {...messages.detailsTabStatsOtherComponents} />}
+          count={otherBlocksCount}
+        />
+      )}
+    </Stack>
+  );
+};
+
+interface CollectionDetailsProps {
+  library: ContentLibrary,
+  collectionId: string,
+}
+
+const CollectionDetails = ({ library, collectionId }: CollectionDetailsProps) => {
+  const intl = useIntl();
+  const { showToast } = useContext(ToastContext);
+
+  const updateMutation = useUpdateCollection(library.id, collectionId);
+  const { data: collection } = useCollection(library.id, collectionId);
```

Comment:
I'm not thrilled that we're reading from the REST API here, while everything else reads from Meilisearch. I would much prefer we carefully designed a clean way to address caching/performance issues across the app instead, otherwise we'll be chasing caching issues everywhere. cf [discussion](https://github.com/openedx/frontend-app-authoring/pull/1281#discussion_r1767726170).

But will let @bradenmacdonald weigh in on whether this matters right now or not.

Reply:
I put a comment in the discussion about why I had to change it here.

Reply:
@rpenido's reasoning makes sense to me and I think the general pattern is: Meilisearch for search results (keywords or faceted filtering are involved), and REST APIs for everything else. I disagree that "everything else" reads from Meilisearch - any of the sidebars should be loading their content from the various REST APIs. The "tags" widget loads from the object tags REST API, the library sidebar loads from the REST API, the editors load from the REST API, and so on. Any time you load a single "thing", it's best to use the REST API because we don't need any Meilisearch features in that case, and it is more likely to be up to date.